js/canvas.js

```diff
@@ -0,0 +1,367 @@
+            nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height;
+          break;
+        case 'bottom-left':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height;
+          break;
+        case 'left-bottom':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY =
+            nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height - draggingAndStickingRect.height;
+          break;
+        case 'left-top':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x - draggingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y;
+          break;
+        case 'right-top':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width;
+          draggingAndStickingRect.stickedY = nearestStandingAndStickingRect.y;
+          break;
+        case 'right-bottom':
+          draggingAndStickingRect.stickedX = nearestStandingAndStickingRect.x + nearestStandingAndStickingRect.width;
+          draggingAndStickingRect.stickedY =
+            nearestStandingAndStickingRect.y + nearestStandingAndStickingRect.height - draggingAndStickingRect.height;
+          break;
+
+        default:
+          break;
+      }
+
+      const { stickedX, stickedY } = draggingAndStickingRect;
+
+      draggingAndStickingRect.arrayOfPrevStickedCoordinates.push({ stickedX, stickedY });
+    }
+
+    if (arrayOfStandingAndStickingRects.length) {
+      arrayOfStandingAndStickingRects.forEach(rect => {
+        rect.stickedX = rect.x;
+        rect.stickedY = rect.y;
+      });
+    }
+  }
+
+  getCurrentXandY(e) {
+    return { currentX: e.clientX - this.offsetX, currentY: e.clientY - this.offsetY };
+  }
+
+  removeStickedCoordinates() {
+    arrayOfRectangles.forEach(rect => {
+      if (!rect.isSticked) {
+        delete rect.stickedX;
+        delete rect.stickedY;
+      }
+    });
+  }
+
+  changeCoordinatesOfRectangleIfHaveStickyCoordinates() {
+    arrayOfRectangles.forEach(rect => {
+      if (rect.stickedX && rect.stickedY) {
+        rect.x = rect.stickedX;
+        rect.y = rect.stickedY;
+      }
+    });
+  }
+
+  mouseDownHandler(e) {
+    const { currentX, currentY } = this.getCurrentXandY(e);
+
+    for (let i = 0; i < arrayOfRectangles.length; i++) {
+      if (
+        currentX >= arrayOfRectangles[i].x &&
+        currentX <= arrayOfRectangles[i].x + arrayOfRectangles[i].width &&
+        currentY >= arrayOfRectangles[i].y &&
+        currentY <= arrayOfRectangles[i].y + arrayOfRectangles[i].height
```

Comment:
more than 3 condition in `if` statement... suggest move condition in separate method/function